### PR TITLE
Add reference to LLM evaluation page in Prompt Engineering UI docs

### PR DESCRIPTION
**Pull request by daniellok-db**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/daniellok-db/mlflow/pull/10279?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/10279/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 10279
```

</p>
</details>

### What changes are proposed in this pull request?

This PR edits the Prompt Engineering UI docs to resolve the ambiguity of the words "evaluate" and "score" in Step 12. This section really only instructs users on how to generate predictions, not to generate metric scores. However, we do have a page specifically on LLM evaluation, so I added another step to reference it.

### How is this PR tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests
Screenshot 2023-11-02 at 1 21 36 PM

https://github.com/mlflow/mlflow/assets/148037680/71a7d290-5e5f-424a-bc06-d13d4cc7e4c1

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [x] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: docs/source/llms/prompt-engineering/index.rst [hunk 1]
```diff
@@ -248,11 +248,11 @@ as follows:
 
 .. _quickstart-score:
 
-Step 12: Score or deploy the best configuration programmatically
```
Comment (daniellok-db) [inline docs/source/llms/prompt-engineering/index.rst:251]: Removed `deploy the best configuration` as the section doesn't really deal with it (though the relevant section is linked below). I can add it back if needed!

File: docs/source/llms/prompt-engineering/index.rst [hunk 2]
```diff
@@ -293,6 +293,13 @@ can score the corresponding MLflow Model in a Python environment of your choosin
    For more information about deployment for real-time serving with MLflow,
    see the :ref:`instructions below <deploy-prompt-serving>`.
 
+Step 13: Perform metric-based evaluation of your model's outputs
+----------------------------------------------------------------
+If you'd like to assess your model's performance on specific metrics, MLflow provides the :py:func:`mlflow.evaluate()`
```
Comment (chenmoneygithub) [inline docs/source/llms/prompt-engineering/index.rst:298]: Maybe we should inline some code showing how to evaluate the model that loaded from Step 12 with `mlflow.evaluate()`, cross referencing a big doc isn't very user friendly.

Comment (BenWilson2) [reply docs/source/llms/prompt-engineering/index.rst:298]: I think a very quick example would work well here too :)

Comment (daniellok-db) [reply docs/source/llms/prompt-engineering/index.rst:298]: Will add one in soon! I just discovered a couple of bugs in the prompt engineering UI code that makes it difficult to construct an example at the moment, but working to get those fixed haha

**Comment by daniellok-db** [comment]
cc @BenWilson2 @chenmoneygithub added an example, let me know how it looks!

**Review by BenWilson2 — APPROVED** [review]
LGTM! Great changes :)

**Pull request merged**